src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -301,14 +301,18 @@ private async Task StopContainerAsync(IExecutionContext executionContext, Contai
             {
                 if (!container.IsJobContainer)
                 {
-                    // Print logs for service container jobs (not the "action" job itself b/c that's already logged).
-                    executionContext.Output($"Print service container logs: {container.ContainerDisplayName}");
-
-                    int logsExitCode = await _dockerManager.DockerLogs(executionContext, container.ContainerId);
-                    if (logsExitCode != 0)
-                    {
-                        executionContext.Warning($"Docker logs fail with exit code {logsExitCode}");
-                    }
+                    var healthcheck = await Healthcheck(executionContext, container);
+                    if (string.Equals(healthcheck, "healthy", StringComparison.OrdinalIgnoreCase)){
```

Comment:
What if the service container becomes unhealthy in the middle of the job? I think we still want to get the log for them.